lrlex/src/lib/ctbuilder.rs

```diff
@@ -79,6 +80,26 @@ pub enum RustEdition {
     Rust2021,
 }
 
+pub trait LexErrorHandler<LexerTypesT>
+where
+    LexerTypesT: LexerTypes,
+    usize: num_traits::AsPrimitive<LexerTypesT::StorageT>,
+{
+    /// Called with the lexers filename
+    fn lexer_path(&mut self, filename: &Path);
+    /// Called with the lexers source contents.
+    fn lexer_src(&mut self, src: &str);
+    fn on_lex_build_error(&mut self, errors: Box<[LexBuildError]>);
+    fn missing_in_lexer(&mut self, missing: &HashSet<String>);
+    fn missing_in_parser(&mut self, missing: &HashSet<String>);
```

Comment:
It is probably worth thinking about whether we can do better than just `HashSet<String>` for `missing_in_*`.

I.e. for `missing_in_lexer`, we can perhaps give a `PIdx` to the parsers production referencing the token.
It would probably need to be given a `YaccGrammar`, and a `YaccGrammarAST` to go from there to spans.
I *think* this is okay to `lrlex`, because it depends on `lrpar`, and overall is where this check occurs currently.

And for `missing_in_parser`, we have `LRNonStreamingLexerDef::get_rule_by_name`, but we aren't passing in a `LRNonStreamingLexerDef` here.

Reply:
> It is probably worth thinking about whether we can do better than just HashSet<String> for missing_in_*.

Yes I tend to agree.

> Not sure the above is currently actionable with the current signature of set_rule_ids though.

I also think I agree with that. I don't immediately have a good idea as to whether we can come up with a better API for `set_rule_ids` or not, but I don't object to changing it, because it's not a function that we expect many people to use directly.